Create Docker instruction

Dmitriy-DevOps-trainee opened the pull request.

Dmitriy-DevOps-trainee committed:
Create Docker instruction

Dmitriy-DevOps-trainee commented:
@cakemanator
@yegorgeorgiyev
@shlepanets
@nikita-safonov

Cakemanator commented on Docker instruction:
```diff
@@ -0,0 +1,14 @@
+
+# After downloading images. Follow this steps:
+sudo docker network create hadoop-net
+sudo docker build -f Dockerfile -t hadoop-headnode:1.0 .    
```
Cannot find the Dockerfiles

Cakemanator reviewed (changes requested):
Changes need to be done

Dmitriy-DevOps-trainee committed:
Update Docker instruction

Cakemanator commented on Docker instruction:
```diff
@@ -0,0 +1,129 @@
+MAINTAINER Dima <dimitry.vlasov@gmail.com>
+
+ARG HADOOP_HOME="/usr/local/hadoop/current"
+ENV HADOOP_HOME="/usr/local/hadoop/current"
```
Why define that 2 times?

Dmitriy-DevOps-trainee replied:
ARG variables can be used to set default settings for ENV variables from the command line during the image build process.
That's why I wrote these lines. And it works. I didn't find any mistakes when creating images and then checking the browser.

Cakemanator replied:
But there is no connections between `ARG` and `ENV`.
If you want to set `ENV` from `ARG` then:
```
ARG HADOOP_HOME="/usr/local/hadoop/current"
ENV HADOOP_HOME="$HADOOP_HOME"
```

Cakemanator commented on Docker instruction:
```diff
@@ -0,0 +1,129 @@
+    chown hdfs:hadoop /usr/local/hadoop/current/logs && \
+    chmod g+w /usr/local/hadoop/current/logs
+
+RUN  mkdir -p /opt/mount{1,2}/namenode-dir && \
```
Can we optimize the amount of `RUN` instructions? What will it give to us?

Dmitriy-DevOps-trainee replied:
You mean to reduce the number of command RUN.  I'm sure we can. I think that this optimization will affect the performance. That's my first experience in Docker and I'm trying to follow step by step, layer by layer.

Cakemanator replied:
Not only performance but image size as well.
Please do that (I mean `RUN` optimization)

Cakemanator commented on Docker instruction:
```diff
@@ -0,0 +1,129 @@
+COPY start_wn.sh /opt/start.sh
+
+USER root
+CMD ["/bin/bash", "/opt/start.sh"]
```
Same for this dockerfile

Dmitriy-DevOps-trainee replied:
Check prevoius comments.

Dmitriy-DevOps-trainee committed:
Update Docker instruction

I have combined some run commands. As far as I know, doing the whole image with only one RUN is not the best practice.
